src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,261 @@
+
+                // Write line to table (plus new line char).
+                writer.write(csvValues);
+
+                // Count number of CSV rows changed.
+                if (!originalValue.equals(transformedValue)) {
+                    modifiedRowCount++;
+                }
+            } // End of iteration over each row.
+            csvReader.close();
+            writer.flush();
+
+            // Copy csv input stream into the zip file, replacing the existing file.
+            try (
+                // Modify target zip file that we just read.
+                FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, null);
+                // Stream for file copy operation.
+                InputStream inputStream =  new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
+            ) {
+                Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+                Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+                zipTarget.feedTransformResult.tableTransformResults.add(
+                    new TableTransformResult(tableName, 0, modifiedRowCount, 0)
+                );
+            }
+
+            // Replace original zip file with temporary working zip file.
+            // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
+            Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
+            LOG.info("Field normalization transformation successful, {} row(s) changed.", modifiedRowCount);
+        } catch (Exception e) {
+            status.fail("Unknown error encountered while transforming zip file", e);
+        }
+    }
+
+    /**
+     * Converts the provided string to Title Case, accommodating for capitalization exceptions
+     * and separator characters that may be immediately precede
+     * (without spaces) text that we need to capitalize.
+     */
+    public String convertToTitleCase(String inputString) {
+        // Return at least a blank string.
+        if (StringUtils.isBlank(inputString)) {
+            return "";
+        }
+
+        String result = WordUtils.capitalizeFully(inputString, SEPARATORS);
+
+        // Exceptions should remain as specified (e.g. acronyms).
+        for (Substitution substitution : getCapitalizationSubstitutions()) {
+            result = substitution.replaceAll(result);
+        }
+
+        return result;
+    }
+
+    /**
+     * Performs substitutions defined for this transform, and returns the result.
+     */
+    public String performSubstitutions(String inputString) {
+        if (substitutions != null && substitutions.size() != 0) {
+            String result = inputString;
+            for (Substitution substitution : substitutions) {
+                result = substitution.replaceAll(result);
+            }
+            return result;
+        }
+        return inputString;
+    }
```

Comment:
Sorry I missed this earlier, but I think a cleaner, less nested approach for this method would be:
```suggestion
   public String performSubstitutions(String inputString) {
        if (substitutions == null) return inputString;
        String result = inputString;
        for (Substitution substitution : substitutions) {
            result = substitution.replaceAll(result);
        }
        return result;
    }
```